DA/Internal/Desugar.hs

```diff
@@ -8,6 +8,7 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE ConstraintKinds #-}
 {-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
Given that we deliberately added the proxy to the other typeclasses to avoid having to enable this, I’d suggest to keep the proxy here as well and not turn this on.